node/src/components/consensus/era_supervisor.rs

```diff
@@ -591,6 +622,64 @@ where
     }
 }
 
+/// Returns an era ID in which the booking block for `era_id` lives in, if we can use it.
+/// We can't use it if it is:
+/// * before Genesis
+/// * before upgrade
+/// * before emergency restart
+/// In those cases, returns `None`.
+fn valid_booking_block(
+    era_id: &EraId,
+    auction_delay: u64,
+    last_activation_point: EraId,
+) -> Option<EraId> {
+    let after_booking_era_id = era_id
+        .saturating_sub(auction_delay)
+        .max(last_activation_point);
+    if after_booking_era_id.is_genesis() {
```

Comment:
Isn't that redundant? If it's genesis, it will also be `== last_activation_point`.

Reply:
I think so, yes. I just wasn't sure (too tired).

Reply:
Removed in [`275d960` (#1068)](https://github.com/CasperLabs/casper-node/pull/1068/commits/275d960613df12b7732d16cf8fa6cc96f5b1270c)